core/events/utils.js

```diff
@@ -365,7 +370,9 @@ const filter = function(queueIn, forward) {
     if (!event.isNull()) {
       // Treat all UI events as the same type in hash table.
       const eventType = event.isUiEvent ? UI : event.type;
-      const key = [eventType, event.blockId, event.workspaceId].join(' ');
+      // TODO(#5927): Ceck whether `blockId` exists before accessing it.
+      const blockId = /** @type {BlockMove} */ (event).blockId;
```

Comment:
This is actually for a Closure Compiler error, rather than a Typescript error. It's triggered by changing the base class to an ES6 class, which caused stricter type checks all around.

I filed https://github.com/google/blockly/issues/5927 to look into this because I couldn't get it to work by just adding `blockId` to the base class--there's an inconsistency in how `blockId` is annotated across different subclasses that pops up when I add it to the base class.

What I actually want to say if: in this hash, use `undefined` for the `blockId` if it doesn't exist on this class, and otherwise use whatever value is found. But I can't check whether this object has that property in a way that Closure compiler agrees with.

Reply:
I don't think that `Abstract` should necessarily have any knowledge of block IDs, because not all events are associated with blocks. Eg [zoom click events](https://developers.google.com/blockly/guides/configure/web/events#blocklyeventsclick). Could `blockId` be added to the `BlockBase` event instead, and then that type can be used in the ternary?

Reply:
I agree with Beka's solution as the long-term solution that should be investigated as part of #5927. I think for the short-term bandaid solution I'm saying I would prefer to do

```
const blockId = /** @type {*} */ (event).blockId;
```

which I think should accomplish the same thing (silence the closure error) but be easier to find as a problem in the future if 5927 isn't picked up relatively soon. I say easier to find because I assume this would get translated in TS to a cast as `any` and I know for a fact that the eslint config we use has a setting to make that an error (we have it as such in samples) and I assume there might be an equivalent rule for closure.

Since we don't mind at all if blockId is undefined, we don't actually *need* the type checking here. I think listing out the individual types is actually more fragile in this case because if we forget one type then those events will not work properly. Therefore I don't think we should go with that suggestion even in the short-term band-aid solution.

Reply:
I'll swap it to `{*}` for this PR. And yes, down the line the typescript compiler will force me to sort this out.